modules/generative-palm/clients/palm.go

```diff
@@ -128,34 +125,115 @@ func (v *palm) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt
+}
+
+func (v *palm) parseGenerativeAIApiResponse(statusCode int, bodyBytes []byte) (*generativemodels.GenerateResponse, error) {
+	var resBody generateMessageResponse
+	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	if err := v.checkResponse(statusCode, resBody.Error); err != nil {
+		return nil, err
+	}
+
+	if len(resBody.Candidates) > 0 {
+		return v.getGenerateResponse(resBody.Candidates[0].Content)
+	}
+
+	return &generativemodels.GenerateResponse{
+		Result: nil,
+	}, nil
+}
+
+func (v *palm) parseResponse(statusCode int, bodyBytes []byte) (*generativemodels.GenerateResponse, error) {
 	var resBody generateResponse
 	if err := json.Unmarshal(bodyBytes, &resBody); err != nil {
 		return nil, errors.Wrap(err, "unmarshal response body")
 	}
 
-	if res.StatusCode != 200 || resBody.Error != nil {
-		if resBody.Error != nil {
-			return nil, fmt.Errorf("connection to Google PaLM failed with status: %v error: %v",
-				res.StatusCode, resBody.Error.Message)
-		}
-		return nil, fmt.Errorf("connection to Google PaLM failed with status: %d", res.StatusCode)
+	if err := v.checkResponse(statusCode, resBody.Error); err != nil {
+		return nil, err
 	}
 
 	if len(resBody.Predictions) > 0 && len(resBody.Predictions[0].Candidates) > 0 {
-		content := resBody.Predictions[0].Candidates[0].Content
-		if content != "" {
-			trimmedResponse := strings.Trim(content, "\n")
-			return &generativemodels.GenerateResponse{
-				Result: &trimmedResponse,
-			}, nil
-		}
+		return v.getGenerateResponse(resBody.Predictions[0].Candidates[0].Content)
 	}
 
 	return &generativemodels.GenerateResponse{
 		Result: nil,
 	}, nil
 }
 
+func (v *palm) getGenerateResponse(content string) (*generativemodels.GenerateResponse, error) {
+	if content != "" {
+		trimmedResponse := strings.Trim(content, "\n")
+		return &generativemodels.GenerateResponse{
+			Result: &trimmedResponse,
+		}, nil
+	}
+
+	return &generativemodels.GenerateResponse{
+		Result: nil,
+	}, nil
+}
+
+func (v *palm) checkResponse(statusCode int, palmApiError *palmApiError) error {
+	if statusCode != 200 || palmApiError != nil {
+		if palmApiError != nil {
+			return fmt.Errorf("connection to Google PaLM failed with status: %v error: %v",
+				statusCode, palmApiError.Message)
+		}
+		return fmt.Errorf("connection to Google PaLM failed with status: %d", statusCode)
+	}
+	return nil
+}
+
+func (v *palm) useGenerativeAIEndpoint(apiEndpoint string) bool {
+	return apiEndpoint == "generativelanguage.googleapis.com"
+}
+
+func (v *palm) getPayload(useGenerativeAI bool, prompt string, settings config.ClassSettings) any {
+	if useGenerativeAI {
+		input := generateMessageRequest{
+			Prompt: &generateMessagePrompt{
+				Messages: []generateMessage{
+					{
+						Content: prompt,
+					},
+				},
+			},
+			Temperature:    settings.Temperature(),
+			TopP:           settings.TopP(),
+			TopK:           settings.TopK(),
+			CandidateCount: 1,
```

Comment:
This probably answers my above question 🙂

Reply:
yes! :)